ecosystem/typescript/sdk_v2/src/client/core.ts

```diff
@@ -67,10 +72,19 @@ export async function aptosRequest<Req, Res>(options: AptosRequest): Promise<Apt
     url: fullEndpoint,
   };
 
+  // to support both fullnode and indexer responses,
+  // check if it is an indexer query, and adjust response.data
+  if (aptosConfig.isIndexerRequest(url)) {
+    // errors from indexer
+    if ((result.data as any).errors) {
+      throw new AptosApiError(options, result, response.data.errors[0].message ?? "Generic Error");
```

Comment:
We should probably find a better way than generic error, it's very confusing.  Let's follow up in a future PR

Reply:
you mean the "Generic Error"? tbh, I don't think it should even get to there - this is for the very rare case where the response doesnt have a message prop, which means something weird happened lol